### PR TITLE
Create .editorconfig

### PR DESCRIPTION
**Pull request by Saibamen**
## Purpose

Add .editorconfig file to automate some files-params while coding

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+[*]
+end_of_line = crlf
+insert_final_newline = true
+charset = utf-8
+indent_size = 4
+indent_style = space
+trim_trailing_whitespace = true
+
+[*.{json,yml,md}]
```
Comment (HBelusca) [inline .editorconfig:14]: Do we use json/yml in our codebase?

Comment (SergeGautherie) [reply .editorconfig:14]: yml is used for appveyor, gitpod and travis, at least.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+[*]
+end_of_line = crlf
```
Comment (ThFabba) [inline .editorconfig:7]: This is not accurate, most files should use native line endings. See `.gitattributes`.

Comment (HBelusca) [reply .editorconfig:7]: According to
https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#end_of_line
there is not yet a value to translate "native line endings"
so I will remove it from the editorconfig.

Comment (HBelusca) [reply .editorconfig:7]: Fixed by commit deebc5848e4e1cff9cbf3706c5ea370076eeb160.

**Pull request merged**